packages/core/src/app/payment/paymentMethod/PaymentMethodV2.tsx

```diff
@@ -70,7 +70,17 @@ const PaymentMethodContainer: ComponentType<
         type: method.type,
     });
 
-    if (!ResolvedPaymentMethod || method.gateway === PaymentMethodId.Mollie) {
+    const shouldUseV1 = (method: PaymentMethod, checkoutState: CheckoutSelectors) => {
+        const useSquareOnV1 =
+            method.id === PaymentMethodId.SquareV2 &&
+            !checkoutState.data.getConfig()?.checkoutSettings.features[
+                'PROJECT-4113.squarev2_web_payments_sdk'
+            ];
+
+        return method.gateway === PaymentMethodId.Mollie || useSquareOnV1;
```

Comment:
How is squareV1 connected to Mollie?

Reply:
It's `SquareOnPaymentMethodV1` not `SquareV1` I changed the name for clarification. We could say that are connected because they have the need to use the 'old' Payment Method component instead the new `V2` under certain circumstances. See #1086